flang/lib/Lower/Mangler.cpp

```diff
@@ -67,13 +67,19 @@ Fortran::lower::mangle::mangleName(fir::NameUniquer &uniquer,
                                    const Fortran::semantics::Symbol &symbol) {
   // Resolve host and module association before mangling
   const auto &ultimateSymbol = symbol.GetUltimate();
-  auto symbolName = ultimateSymbol.name();
+
+  auto symbolNameString = ultimateSymbol.name();
+  auto symbolName = toStringRef(symbolNameString);
```

Comment:
So this conversion is nearly as cheap as a nop. :)

Reply:
Good point, I will join the two lines,  `symbolNameString` is not used, I just thought it was a string and I had to keep it alive so that  the StringRef stayed valid. But as you said, its a CharBlock...